sequence_processing_pipeline/Pipeline.py

```diff
@@ -347,11 +348,43 @@ def get_project_info(self):
 
     @staticmethod
     def is_mapping_file(mapping_file_path):
+        '''
+        Returns true if file is a mapping file.
+        '''
         try:
             Pipeline._validate_mapping_file(mapping_file_path)
             return True
-        except PipelineError:
-            return False
+        except PipelineError as e:
+            # we want to distinguish between a file that is clearly not a
+            # mapping file e.g. sample-file, and a mapping-file that is perhaps
+            # missing a column or has duplicate sample-names.
+            messages = ['duplicate sample-names detected:', 'missing columns:']
+            for message in messages:
+                if str(e).startswith(message):
+                    return True
+
+        return False
+
+    @staticmethod
+    def is_sample_sheet(sample_sheet_path):
+        '''
+        Returns True if file follows basic sample-sheet format.
+        '''
+
+        # Check to see if the file begins w/[Header].
+        # Ignoring any legacy comments.
+        with open(sample_sheet_path, 'r') as f:
+            line = f.readline()
+            while line:
+                if line.startswith('#'):
+                    line = f.readline()
+                else:
+                    break
+
+            if line.startswith('[Header]'):
+                return True
```

Comment:
Should these lines be outside the `with open...`

Reply:
For this I would request to everything under the with open() as it makes it easy to process individual lines for potential comments and it allows us to not have to read the entire file. Having the entire while loop underneath the with open() allows us to do this.